src/vision/status_quo/niklaus_lost_his_context.md

```diff
@@ -0,0 +1,99 @@
+
+If you would like to expand on this story, or adjust the answers to the FAQ, feel free to open a PR making edits (but keep in mind that, as they reflect peoples' experiences, status quo stories [cannot be wrong], only inaccurate). Alternatively, you may wish to [add your own status quo story][htvsq]!
+
+## The story
+
+Niklaus heard about a project to reimplement a deprecated browser plugin using Rust and WASM. This old technology had the ability to load resources over HTTP; so it makes sense to try and implement that functionality using the Fetch API. He looks up the documentation of `web_sys` and realizes that he needs to take a thing called a `Future` out of the call to `fetch`, do some things to it in Rust, and then send the future off to `spawn_local` as such:
+
+```use wasm_bindgen_futures::spawn_local;
+use web_sys::window;
+
+fn load_image(src: &url) {
+    spawn_local(async move {
+        let url = src.to_string();
+        let request = //request construction in Rust is actually pretty verbose
+                      //just pretend it's here
+        window().unwrap().fetch_with_request(&request).await;
+        log::error!("It worked");
+    });
+}
+```
+
+Niklaus adds calls to `load_image` where appropriate, sees the message pop up in the console, and figures it's time to now actually do something to that loaded image. At this point, it's important to note that many parts of this project pass around "contexts" - bundles of borrowed state that various parts of the reimplementation are free to alter. (This is necessary because this plugin had multiple scripting runtimes in it.) In synchronous code, this was perfectly fine. However, when we start integrating the function into the player...
+
+```use crate::{Index, Context}
+use wasm_bindgen_futures::spawn_local;
+use web_sys::window;
+use crate::parse_png;
+
+fn load_image(src: &url, inside_of: Index<Node>, context: &mut Context<'_>) {
+    spawn_local(async move {
+        let url = src.to_string();
+        let request = //still elided
+        let data = window().unwrap().fetch_with_request(&request).await.unwrap().etc.etc.etc;
+        let image = parse_png(data, context);
+        context.display_tree.add_child_node(inside_of, image);
+    });
+}
+```
+
+Well, that stopped compiling. After looking through far too many compiler error messages, Niklaus realizes that the problem is the context itself. You see, whenever you close over some borrowed state, your closure's lifetime is bounded by that borrow. This is reasonable for closures, as there are plenty of contexts where you would pass such a closure and have it called immediately. However, the whole point of futures is that they are *not* called immediately - you hold them inside of a thing called an "executor", which typically needs futures that own their contents.
```

Comment:
...might be nice to include an example of those error messages...but maybe too annoying to reproduce :)